src/ncp/CMakeLists.txt

```diff
@@ -84,7 +84,7 @@ set(COMMON_SOURCES
 set(OT_NCP_VENDOR_HOOK_SOURCE "" CACHE STRING "set vendor hook source file for NCP")
 if(OT_NCP_VENDOR_HOOK_SOURCE)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_ENABLE_NCP_VENDOR_HOOK=1")
-    list(APPEND COMMON_SOURCES ${OT_NCP_VENDOR_HOOK_SOURCE})
+    list(APPEND COMMON_SOURCES ${OT_NCP_VENDOR_HOOK_SOURCE_DIR}${OT_NCP_VENDOR_HOOK_SOURCE})
```

Comment:
nit: add a space?
```suggestion
    list(APPEND COMMON_SOURCES ${OT_NCP_VENDOR_HOOK_SOURCE_DIR} ${OT_NCP_VENDOR_HOOK_SOURCE})
```

Reply:
I don't think so, it's about string parts containing path to file concatenation. I added OT_NCP_VENDOR_HOOK_SOURCE_DIR part to simplify user configuration part. OT_NCP_VENDOR_HOOK_SOURCE may contain only file name and OT_NCP_VENDOR_HOOK_SOURCE_DIR is full path to the file directory passed e.g. by cmake CMAKE_CURRENT_SOURCE_DIR.